Precompile failures on julia nightly

```
Failed to precompile Flux [587475ba-b771-5e3f-ad9e-33799f191a9c] to "/home/runner/.julia/compiled/v1.11/Flux/jl_hOZ50V".
ERROR: LoadError: AssertionError: precompile(Tuple{typeof((which(__broadcast, (Any, Size, Tuple{Vararg{Size}}, Vararg{Any}))).generator.gen), Any, Any, Any, Any, Any, Any})
Stacktrace:
 [1] _precompile_()
   @ StaticArrays ~/.julia/packages/StaticArrays/t0DdQ/src/precompile.jl:25
 [2] top-level scope
   @ ~/.julia/packages/StaticArrays/t0DdQ/src/StaticArrays.jl:139
 [3] include(mod::Module, _path::String)
   @ Base ./Base.jl:489 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{…}, dl_load_path::Vector{…}, load_path::Vector{…}, concrete_deps::Vector{…}, source::String)
   @ Base ./loading.jl:2216
 [5] top-level scope
   @ stdin:3
Some type information was truncated. Use `show(err)` to see complete types.
in expression starting at /home/runner/.julia/packages/StaticArrays/t0DdQ/src/StaticArrays.jl:1
in expression starting at stdin:3
ERROR: LoadError: Failed to precompile StaticArrays [90137ffa-7385-5640-81b9-e5203721[818](https://github.com/r3tex/ObjectDetector.jl/actions/runs/5577159290/jobs/10189481031?pr=88#step:4:822)2] to "/home/runner/.julia/compiled/v1.11/StaticArrays/jl_HtO2XA".
```

--- a/src/precompile.jl
+++ b/src/precompile.jl
@@ -21,13 +21,15 @@ function _precompile_()
         end
     end
 
-    # Some expensive generators
-    @assert precompile(Tuple{typeof(which(__broadcast,(Any,Size,Tuple{Vararg{Size}},Vararg{Any},)).generator.gen),Any,Any,Any,Any,Any,Any})
-    @assert precompile(Tuple{typeof(which(_zeros,(Size,Type{<:StaticArray},)).generator.gen),Any,Any,Any,Type,Any})
-    @assert precompile(Tuple{typeof(which(_mapfoldl,(Any,Any,Colon,Any,Size,Vararg{StaticArray},)).generator.gen),Any,Any,Any,Any,Any,Any,Any,Any})
-
+    # TODO: These fail to precompile on v1.11 pre-release
+    if VERSION < v"1.11.0-0"
+        # Some expensive generators
+        @assert precompile(Tuple{typeof(which(__broadcast,(Any,Size,Tuple{Vararg{Size}},Vararg{Any},)).generator.gen),Any,Any,Any,Any,Any,Any})
+        @assert precompile(Tuple{typeof(which(_zeros,(Size,Type{<:StaticArray},)).generator.gen),Any,Any,Any,Type,Any})
+        @assert precompile(Tuple{typeof(which(_mapfoldl,(Any,Any,Colon,Any,Size,Vararg{StaticArray},)).generator.gen),Any,Any,Any,Any,Any,Any,Any,Any})
+    end
     # broadcast_getindex
-    for m = 0:5, n = m:5 
+    for m = 0:5, n = m:5
         @assert precompile(Tuple{typeof(broadcast_getindex),NTuple{m,Int},Int,CartesianIndex{n}})
     end
 end

